start

```diff
@@ -48,14 +48,45 @@ parse_args () {
     esac
 }
 
-check_files_dont_exist () {
-    if [ -e WORKSPACE ] || [ -e BUILD ] || [ -e BazelExample.hs ]
+check_dir () {
+    if [ -d "$1" ] # we can't write a file when a directory of that name exists
+    then
+        echo "STOP: There's a directory named $1 but we want to write a file with the same name. Please delete or rename the $1 directory." >&2
+        exit 1
+    fi
+}
+
+check_alt () {
+    check_dir $2
+    if [ -f "$1" ]
+    then
+        echo "STOP: There's a $1 file but we intend to write a $2 file. When both exist, bazel will pick the one with the .bazel extension, ignoring the other one. Please delete the $1 file." >&2
+        exit 1
+    fi
+}
+
+check_clash () {
+    check_dir $1
+    if [ -e "$1" ]
     then
-        echo "Current directory already has WORKSPACE and/or BUILD and/or BazelExample.hs files." >&2
+        echo "STOP: The current directory already has a $1 file and we don't want to overwrite it." >&2
         exit 1
     fi
 }
 
+check_files_dont_exist () {
+    # A BUILD.bazel file takes precedence over a BUILD file and likewise
+    # a WORKSPACE.bazel file takes precedence over a WORKSPACE file. We write
+    # BUILD.bazel and WORKSPACE files. Seems an odd mix.
```

Comment:
```suggestion
    # BUILD.bazel and WORKSPACE files.
    # Some Bazel tooling may fail on WORKSPACE.bazel files,
    # e.g. https://github.com/bazelbuild/bazel-gazelle/issues/678
```
Clashes with `BUILD` are relatively common, while clashes with `WORKSPACE` aren't. I assume that is why most Bazel projects still use `WORKSPACE`. It seems that some Bazel tooling also has trouble with `WORKSPACE.bazel`, e.g. the cited issue. So, using `WORKSPACE` is the safer option for now.